data/transactions/logic/tracer.go

```diff
@@ -145,7 +145,11 @@ type EvalTracer interface {
 	// groupIndex refers to the index of the transaction in the transaction group that was just executed.
 	// ad is the ApplyData result of the transaction; prefer using this instead of
 	// ep.TxnGroup[groupIndex].ApplyData, since it may not be populated at this point.
-	AfterTxn(ep *EvalParams, groupIndex int, ad transactions.ApplyData, evalError error)
+	//
+	// If GranularEval is enabled by the evaluator (specified either in logic.EvalParams or ledger's
+	// BlockEvaluator), deltas is the ledgercore.StateDelta changes that occurred because of this
+	// transaction. Otherwise, this argument is nil.
```

Comment:
Do we need to talk about `GranularEval` here? It seems like that is something that the tracing facility needs to set up, if a Tracer is added to the block evaluator. That is, `AfterTxn` should always get StateDeltas and need not know about how it works.

Reply:
If I understand correctly, you're advocating for evaluators to always create cows for individual transactions if tracers are present, removing the need for a separate `GranularEval` flag?

My current thoughts are that it would be too much to assume every tracer implementation cares about transaction-level StateDeltas. For example, I imagine simulate would only want StateDeltas at that granularity for execution traces.

What are your thoughts on a new `EvalTracer` method that would indicate whether the tracer wants StateDeltas in `AfternTxn`? Something like `EvalTracer.AfterTxnGranularStateDeltas() bool`?